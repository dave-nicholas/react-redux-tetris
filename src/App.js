import React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { Fragment, push } from 'redux-little-router';
import Button from 'material-ui/Button';
import { withStyles } from 'material-ui/styles';
import Toolbar from 'material-ui/Toolbar';
import Grid from 'material-ui/Grid';
import { teal, purple } from 'material-ui/colors';
import SplashScreen from './features/SplashScreen/SplashScreen';
import SubmitScore from './features/SubmitScore/SubmitScore';
import Game from './features/Tetris/Game';
import Message from './features/SnackBar/Message';

const styles = theme => ({
  root: {
    width: '100%'
  },
  logo: {
    width: '50px',
    cursor: 'pointer'
  },
  title: {
    color: '#fff',
    marginLeft: '150px'
  },
  appBar: {
    background: `linear-gradient(45deg, ${teal[300]} 30%, ${teal[500]} 90%)`
  },
  navButton: {
    marginLeft: theme.spacing.unit * 2,
    color: '#fff',
    background: `linear-gradient(45deg, ${purple[300]} 30%, ${purple[500]} 90%)`
  }
});

const App = ({ push, classes }) => {
  return (
    <div className={classes.root}>
      <Grid container spacing={0} justify="center" alignItems="center">
        <Grid item>
          <Toolbar>
            <Button
              raised
              className={classes.navButton}
              onClick={() => push('/')}
            >
              Home
            </Button>
            <Button
              raised
              className={classes.navButton}
              onClick={() => push('/game')}
            >
              Play
            </Button>
          </Toolbar>
        </Grid>
        <Grid item xs={12}>
          <Grid container justify="center" spacing={0}>
            <Fragment forRoute="/">
              <div>
                <Fragment forRoute="/">
                  <SplashScreen />
                </Fragment>
                <Fragment forRoute="/game">
                  <Game />
                </Fragment>
                <Fragment forRoute="/submit-score">
                  <SubmitScore />
                </Fragment>
              </div>
            </Fragment>
          </Grid>
        </Grid>
      </Grid>
      <Message />
    </div>
  );
};

export default compose(
  withStyles(styles),
  connect(null, {
    push
  })
)(App);
