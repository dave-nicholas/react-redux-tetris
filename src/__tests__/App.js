import React from 'react';
import App from '../App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  const app = shallow(
    <App classes={{ root: '', navButton: '' }} push={location => {}} />
  );
  expect(app).toMatchSnapshot();
});
