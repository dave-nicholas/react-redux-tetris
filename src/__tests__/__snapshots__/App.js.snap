// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withStyles(Connect(App))
    classes={
        Object {
            "navButton": "",
            "root": "",
          }
    }
    push={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "appBar": "Connect-App--appBar-4",
        "logo": "Connect-App--logo-2",
        "navButton": "Connect-App--navButton-5",
        "root": "Connect-App--root-1",
        "title": "Connect-App--title-3",
      },
      "push": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "appBar": "Connect-App--appBar-4",
          "logo": "Connect-App--logo-2",
          "navButton": "Connect-App--navButton-5",
          "root": "Connect-App--root-1",
          "title": "Connect-App--title-3",
        },
        "push": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
